cpp/density/RDF.h

```diff
@@ -32,53 +29,99 @@ class RDF : public util::NdHistogram
     void reset();
 
     //! Compute the RDF
+    /*! Accumulate the given points to the histogram. Accumulation is performed
+     * in parallel on thread-local copies of the data, which are reduced into
+     * the primary data arrays when the user requests outputs.
+     */
     void accumulate(const freud::locality::NeighborQuery* neighbor_query,
                     const vec3<float>* query_points, unsigned int n_query_points,
                     const freud::locality::NeighborList* nlist, freud::locality::QueryArgs qargs);
 
-    //! \internal
-    //! helper function to reduce the thread specific arrays into one array
-    void reduceRDF();
+    //! Reduce thread-local arrays onto the primary data arrays.
+    virtual void reduce();
 
-    //! Implementing pure virtual function from parent class.
-    virtual void reduce()
+    //! Return :code:`thing_to_return` after reducing.
+    template<typename T>
+    T &reduceAndReturn(T &thing_to_return)
     {
-        reduceRDF();
+        if (m_reduce == true)
+        {
+            reduce();
+        }
+        m_reduce = false;
+        return thing_to_return;
     }
 
-    //! Get a reference to the PCF array
+    //! Get the simulation box
+    const box::Box& getBox() const
+    {
+        return m_box;
+    }
+
+    //! Get the positional correlation function.
     const util::ManagedArray<float> &getRDF()
     {
-        return getPCF();
+        return reduceAndReturn(m_pcf);
     }
 
-    //! Get a reference to the r array
-    const util::ManagedArray<float> &getR();
+    //! Get the histogram of bin distances.
+    const util::ManagedArray<unsigned int> &getBinCounts()
+    {
+        return m_histogram.getBinCounts();
+    }
 
     //! Get a reference to the N_r array.
-    /*! Mathematically, m_N_r_array[i] is the average number of points
-     * contained within a ball of radius m_r_array[i]+dr/2 centered at a given
+    /*! Mathematically, m_N_r[i] is the average number of points
+     * contained within a ball of radius getBins()[i+1] centered at a given
      * query_point, averaged over all query_points.
      */
     const util::ManagedArray<float> &getNr()
     {
-        return reduceAndReturn(m_N_r_array);
+        return reduceAndReturn(m_N_r);
+    }
+
+    float getRMax() const
+    {
+        return m_r_max;
     }
 
-    unsigned int getNBins();
+    float getRMin() const
+    {
+        return m_r_min;
+    }
+
+    //! Get bin centers.
+    std::vector<float> getBinCenters() const
+    {
+        // RDFs are always 1D histograms, so we just return the first element.
+        return m_histogram.getBinCenters()[0];
+    }
+
+    //! Return the bin boundaries.
+    std::vector<float> getBins() const
+    {
+        // RDFs are always 1D histograms, so we just return the first element.
+        return m_histogram.getBinBoundaries()[0];
```

Comment:
Tests failed, I think this will fix it.
```suggestion
        return m_histogram.getBinEdges()[0];
```